rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -1113,6 +1113,23 @@ public Case(UUID id, Space prefix, Markers markers, Type type, @Deprecated @Null
             this.body = body;
         }
 
+        public Case withGuard(@Nullable Expression guard) {
+            if (this.guard == guard) {
+                return this;
+            }
+            if (this.guard == null) {
+                List<JRightPadded<J>> caseLabels = ListUtils.mapLast(getPadding().getCaseLabels().getPadding().getElements(), last -> {
+                    if (last != null && Space.EMPTY.equals(last.getAfter())) {
+                        return last.withAfter(Space.SINGLE_SPACE);
+                    }
+                    return last;
+                });
+
+                return new Case(this.padding, this.id, this.prefix, this.markers, this.type, JContainer.build(this.caseLabels.getBefore(), caseLabels, this.caseLabels.getMarkers()), this.statements, this.body, guard);
```

Comment:
Here you would typically just use `JContainer#withElements()`:

```suggestion
                return new Case(this.padding, this.id, this.prefix, this.markers, this.type, this.caseLabels.withElements(caseLabels), this.statements, this.body, guard);
```

But actually, this type of problem we tend to solve by adding logic to the formatter. In this case specifically to `MinimumViableSpacingVisitor`, as this space is required to make the code compile.

Reply:
Also note that `JavaTemplate` should automatically apply the formatter, so the recipe shouldn't have to do that itself.

Reply:
Thanks for the feedback, @knutwannheden !
I adapted the logic to be in the `MinimumViableSpacingVisitor` but it seems that it is not automatically being called. 
I am not using a JavaTemplate here, so that might be it (not for the guard at least, only for the condition). The test added in this PR fails when using the `MinimumViableSpacingVisitor` unless I specifically call it in my test (see commit just made.